src/natcap/invest/urban_mental_health.py

```diff
@@ -0,0 +1,1335 @@
+            },
+            target_path_list=[file_registry['population_aligned']],
+            task_name='Resample population to NDVI resolution')
+
+        target_bounding_box = pygeoprocessing.get_raster_info(
+            file_registry['population_aligned'])["bounding_box"]
+        if not numpy.allclose(numpy.array(target_bounding_box),
+                              numpy.array(delta_ndvi_bbox), rtol=0, atol=1e-6):
+            raise ValueError(
+                "The bounding boxes of the preprocessed population raster and "
+                "delta_ndvi raster are not equal. This occurs when the "
+                "population raster is too small. The bounds of the population "
+                "raster should at least extend across the entire AOI.")
+
+        baseline_cases_task = task_graph.add_task(
+            func=calc_baseline_cases,
+            args=(file_registry['population_aligned'],
+                  args['baseline_prevalence_vector'],
+                  file_registry['baseline_prevalence_raster'],
+                  file_registry['baseline_cases']),
+            target_path_list=[file_registry['baseline_cases']],
+            dependent_task_list=[population_align_task],
+            task_name="calculate baseline cases"
+        )
+
+        preventable_cases_task = task_graph.add_task(
+            func=calc_preventable_cases,
+            args=(file_registry['delta_ndvi'],
+                  file_registry['baseline_cases'],
+                  args['effect_size'],
+                  file_registry['preventable_cases'],
+                  args["aoi_vector_path"],
+                  args['workspace_dir']),
+            target_path_list=[file_registry['preventable_cases']],
+            dependent_task_list=[delta_ndvi_task, baseline_cases_task],
+            task_name="calculate preventable cases"
+        )
+
+        if args['health_cost_rate']:
+            LOGGER.info("Calculating preventable cost")
+            preventable_cost_task = task_graph.add_task(
+                func=calc_preventable_cost,
+                args=(file_registry['preventable_cases'],
+                      args['health_cost_rate'],
+                      file_registry['preventable_cost']),
+                target_path_list=[file_registry['preventable_cost']],
+                dependent_task_list=[preventable_cases_task],
+                task_name="calculate preventable cost"
+            )
+
+        task_graph.join()
+        # TODO ^ is this best way to require prev cost task has been completed?
+        # Can't add as dependent task below as not done if not health_cost_rate
+
+        zonal_stats_inputs = [
+            args['aoi_vector_path'],
+            file_registry['preventable_cases_cost_sum_table'],
+            file_registry['preventable_cases_cost_sum_vector'],
+            file_registry['preventable_cases']]
+
+        zonal_stats_dependent_tasks = [preventable_cases_task]
+
+        if args['health_cost_rate']:
```

Comment:
I _think_ you could move the above `preventable_cost_task` code under this same conditional, then add to the input and dependent task list, and then remove that `task_graph.join()` above. If I'm reading that right!